### PR TITLE
LUCENE-10200: store git revision in the release folder and read it back from buildAndPushRelease

### PR DESCRIPTION
**Pull request by dweiss**

### CODE REVIEW

File: gradle/generation/kuromoji.gradle [hunk 1]
```diff
@@ -73,7 +73,10 @@ configure(project(":lucene:analysis:kuromoji")) {
         // Apply patch via local git.
         project.quietExec {
           workingDir = unpackedDir
-          executable "git" // TODO: better use jgit to apply patch, this is not portable!!!
```
Comment (dweiss) [inline gradle/generation/kuromoji.gradle:76]: Piggybacking this unrelated change. It's only a comment reference to another issue.

**Review by janhoy — APPROVED** [review]
Skimming through, not tested. Agree it is better to keep git rev together with repo in build folder.

**Comment by dweiss** [comment]
I tested it on Linux and it worked fine for me. Applying - we can always correct if something is not right.
